Status filtering

tshuck opened the pull request:
@hatchloyalty/core 

Updates the Offer guide with more information on Visibility, Eligibility and Activation.  Also adds a gatsby package to generate ids for `h1` and `h2` elements so that anchor linking will work again (and fixes existing tutorials anchor links for gatsby-generated pages from markdown).

Adds a "Retrieve Membership Offers" tutorial that describes how to efficiently retrieve membership-offers with the `visible_status` filter and a bunch more info.

Please feel free to read copy closely & suggest edits.

tshuck commented:
So we deploy to production on master merge (this helps Natasha self-serve on blog posts, etc.), but it's sort of annoying since this documentation will not be correct until we deploy the platform to production.  Gonna leave this sitting open until we can think of a better workflow with regard to staging -- please feel free to continue to edit in the meantime.

DLavin23 commented on content/static-pages/guides-and-tutorials.md:
```diff
@@ -19,6 +19,10 @@ How to compile and configure a complex purchase-based marketing initiative in Ha
 
 How to acquire new Memberships in your program with a tailored onboarding.
 
+## [Membership Offers](/tutorials/membership-offers)
```
FYI, this link does not appear to work. 

https://deploy-preview-46--stupefied-cori-df6191.netlify.com/tutorials/membership-offers

tshuck commented:
@DLavin23 Great catch -- link fixed.

Shiggiddie commented on content/static-pages/guides/offers.md:
```diff
@@ -9,73 +9,109 @@ Offers are exposed via an API resource called [Offers](/api/#offers).
+The `activation_status` of `requires_activation` and `does_not_require_activation` apply to all memberships equally and may be returned when calling the Offers api; an `activation_status` of `activated` is only available in the context of a Membership-Offer and will only be returned when calling the Membership-Offers api.
 
-**Possible Values**: `true`, `false`, `null`
+The list below describes each of the possible values, as well as the context (global, and/or per-membership) that the value may be returned in.
```
Change:
> returned in

Ending a sentence in a preposition.

Another formulation may be:
as well as the value's return context (global, and/or per-membership)

Shiggiddie commented on content/static-pages/guides/offers.md:
```diff
@@ -9,73 +9,109 @@ Offers are exposed via an API resource called [Offers](/api/#offers).
-## Fulfillment Count
+## Other attributes
+
+While `visible_status`, `eligible` and `activation_status` are the primary attributes used to understand the relationship between Memberships and Offers, both the Offer and Membership Offer provide additional information that may be used when displaying an offer to a member.
```
Change:
I am just now noticing the inconsistency between "Membership Offer" and "Membership-Offer".  If there is no purpose for this inconsistency, we should make it consistent

Shiggiddie commented on content/static-pages/tutorials/retrieve-membership-offers.md:
```diff
@@ -0,0 +1,366 @@
+
+## Intro
+
+A successful marketing program may consist of offers with a wide variety of configurations; some are meant to engender general engagement with the program and are available to all members, but others may be targeted to a specific group of members based on their behavioral or demographic characteristics. Offers may also require certain actions, like _Activation_, or be constrained by fulfillment l ...
```
Should _Activation_ be a link here?  Just asking, no preference here

Shiggiddie commented on content/static-pages/tutorials/retrieve-membership-offers.md:
```diff
@@ -0,0 +1,366 @@
+
+A marketing program is most effective when it is able to target a member with offers specific to their needs, and is most helpful to a member when it can display only those offers that they're able to engage with. It is wasteful, for instance, to display an offer that a member may not fulfill because they do not belong to a required [custom segment](/guides/custom-segments), or because the offer  ...
+
+The [Membership-Offers](https://developer.hatchloyalty.com/api/#membership-offers) will already display only the offers that belong to the same _Chain_ as the member, but will default to including offers that are discontinued, unpublished, or limited -- as well as those that are available for the member to engage with. Further, some of these offers may require additional prerequisite actions to b ...
```
Change:
> as well as those that are available for the member to engage with.

Ending a sentence in a preposition.

Another formulation:

> as well as those that are available for the member's engagement.

Shiggiddie commented on content/static-pages/tutorials/retrieve-membership-offers.md:
````diff
@@ -0,0 +1,366 @@
+}
+```
+
+Many fewer offers have now been retrieved -- an effective marketing campaign contains some offers that all members may engage with (e.g., receiving a small number of points for each dollar spent), but many that are specifically targeted to Memberships based on their [Audiences](/guides/offers/audiences). All of the offers retrieved in this way may be shown immediately to the member because they a ...
````
> Many fewer offers have now been retrieved

May be better communicated & educational by a tweak:

> We see by the `meta` key's `record_count` value of **12** compared to the previous requests **500** that many fewer offers have now been retrieved.

Shiggiddie replied:
>  retrieved -- an effective

Should really be "--" or a period?

Shiggiddie commented on content/static-pages/tutorials/retrieve-membership-offers.md:
```diff
@@ -0,0 +1,366 @@
+
+Many fewer offers have now been retrieved -- an effective marketing campaign contains some offers that all members may engage with (e.g., receiving a small number of points for each dollar spent), but many that are specifically targeted to Memberships based on their [Audiences](/guides/offers/audiences). All of the offers retrieved in this way may be shown immediately to the member because they a ...
+
+Note, though, that this offer still indicates that it is `eligible: false`. We may want to deal with these offers explicitly by providing specific information to the member as to what they must do in order to be eligible for this offer.
```
personal preference "however" instead of "though" in "Note, though,"

Shiggiddie commented on content/static-pages/tutorials/retrieve-membership-offers.md:
```diff
@@ -0,0 +1,366 @@
+
+### Step 3: _Eligibility_
+
+While an offer that is ineligible for a member may not be engaged with, the [Eligibility](/guide/offers#eligibility) documentation indicates that an ineligible offer (as long as it is otherwise _visible_) requires that the user take a certain prerequisite action before engaging with it. Although _Activation_ is currently the only prerequisite action supported by Hatch, there may be many more in t ...
```
Change:
> While an offer that is ineligible for a member may not be engaged with

This phrase indicates that the offer is not "engage-able" strictly by the use of the words "may not be engaged with".  However later in this same sentence we talk about how a member can engage with an ineligible offer.  I am guessing that "not visible" was intended instead of "ineligible"?

Shiggiddie replied:
> documentation indicates that an ineligible offer (as long as it is otherwise _visible_) requires

I found this difficult to read, a proposed reformulation:

> documentation indicates that as long as an ineligible offer is _visible_ the member must take prerequisite action(s) before the offer is eligible.

Shiggiddie replied:
Change:
>  that the user

Should be "member"

Shiggiddie replied:
> Although _Activation_ is currently the only prerequisite action supported by Hatch, there may be many more in the future. For this reason, it's not helpful to simply filter out all of the ineligible offers -- but instead to inspect the other properties of the offer in order to indicate the desired action.

This feels mostly unnecessary, but may work in one of those "PROTIP"/"CAUTION" sections discussed above

Shiggiddie commented on content/static-pages/tutorials/retrieve-membership-offers.md:
```diff
@@ -0,0 +1,366 @@
+
+This sequence of steps describes the process necessary to fetch relevant offers for a member, and to use the `eligible` and `activation_status` fields to indicate the actions that are required to engage with each offer. It also describes filtering on `activation_status` for special views, and how to reduce request overhead by eagerly-loading marketing_contents and reduce payload size with sparse  ...
+
+These tools are indispensable when engaging a member with the offers they care about, and help ensure that they'll engage with your loyalty program.
```
> These tools are indispensable when engaging a member with the offers they care about

Seems like a good opportunity to add in some kind of Hatch marketing slogan here that we can adopt throughout our documentation.

I imagine something like "the right customer, with the right offer, at the right time" or something like that.

Something much more in @DLavin23 or someone more on the business team to consider?

tshuck commented on content/static-pages/guides/offers.md:
```diff
@@ -9,73 +9,109 @@ Offers are exposed via an API resource called [Offers](/api/#offers).
+  - The offer [requires activation](/guides/offers/constraints#requires-activation) and has not been activated.
 
-## Must Activate
+Because an offer may be ineligible due either to _Visibility_ or a prerequisite action the the member must take, it may not be useful to filter out ineligible offers from view without further consideration. Instead, the `visible_status` (above) or `activation_status` (below) should be considered for ineligible offers.
```
This is a great idea.  I'll change to `Note: because` for now and schedule this for a future cooldown if we don't prioritize it earlier.

tshuck commented:
@Shiggiddie @camneu37 Thank you both for your close-reading and excellent comments.  Did some things:
  - Got rid of dangling prepositions
  - Updated "membership" -> "member" (rule of thumb:  `Membership` when referring to a resource like `Membership-Offer`, `member` when talking about a person)
  - Updated all `Membership Offer` to `Membership-Offer`
  - Pulled out some sections with `Note: <blah blah>`
  - Fixed other odd syntax

tshuck commented:
@DLavin23 We'll need your green checkmark before we can merge this on Monday after the production release goes out.
